pkg/teeth/publish_logs_test.go

```diff
@@ -124,3 +169,141 @@ ORDER BY
+		},
+		{
+			name:     "insert_fails",
+			bqClient: FakeBigQueryClientWithDefaultConfig(make([]*InvocationCommentStatusRecord, 0), errors.New("BOOM"), DefaultBQInsertBatchSize, DefaultBQInsertBatchWait),
+			statuses: []*InvocationCommentStatusRecord{
+				{
+					PullRequestID:  123456789,
+					PullRequestURL: "https://github.com/foo/bar/pull/1",
+					ProcessedAt:    time.Now().UTC(),
+					CommentID:      bigquery.NullInt64{Int64: time.Now().Unix()},
+					Status:         "SUCCESS",
+					JobName:        "job-0",
+				},
+			},
+			wantErrorCount: 1,
+		},
+		{
+			name:     "insert_fails_context_cancelled",
+			bqClient: FakeBigQueryClientWithDefaultConfig(make([]*InvocationCommentStatusRecord, 0), context.Canceled, DefaultBQInsertBatchSize, DefaultBQInsertBatchWait),
+			statuses: []*InvocationCommentStatusRecord{
+				{
+					PullRequestID:  123456789,
+					PullRequestURL: "https://github.com/foo/bar/pull/1",
+					ProcessedAt:    time.Now().UTC(),
+					CommentID:      bigquery.NullInt64{Int64: time.Now().Unix()},
+					Status:         "SUCCESS",
+					JobName:        "job-0",
+				},
+			},
+			wantErrorCount: 1,
+		},
+		{
+			name:     "insert_fails_with_batching",
+			bqClient: FakeBigQueryClientWithDefaultConfig(make([]*InvocationCommentStatusRecord, 0), errors.New("BOOM"), 2, DefaultBQInsertBatchWait),
+			statuses: []*InvocationCommentStatusRecord{
+				{
+					PullRequestID:  123456789,
+					PullRequestURL: "https://github.com/foo/bar/pull/1",
+					ProcessedAt:    time.Now().UTC(),
+					CommentID:      bigquery.NullInt64{Int64: time.Now().Unix()},
+					Status:         "SUCCESS",
+					JobName:        "job-0",
+				},
+				{
+					PullRequestID:  987654321,
+					PullRequestURL: "https://github.com/foo/bar/pull/2",
+					ProcessedAt:    time.Now().UTC(),
+					CommentID:      bigquery.NullInt64{Int64: time.Now().Unix()},
+					Status:         "SUCCESS",
+					JobName:        "job-0",
+				},
+				{
+					PullRequestID:  999999999,
+					PullRequestURL: "https://github.com/foo/bar/pull/3",
+					ProcessedAt:    time.Now().UTC(),
+					CommentID:      bigquery.NullInt64{Int64: time.Now().Unix()},
+					Status:         "SUCCESS",
+					JobName:        "job-0",
+				},
+			},
+			wantErrorCount: 2,
+		},
+	}
+	for _, tc := range tests {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+			ctx := context.Background()
+			err := SaveInvocationCommentStatus(ctx, tc.bqClient, tc.statuses)
+			gotErrCount := 0
+			if err != nil {
+				gotErrCount = len(strings.Split(err.Error(), "\n"))
+			}
+			if gotErrCount != tc.wantErrorCount {
+				t.Errorf("SaveInvocationCommentStatus returned %v, want %d errors", err, tc.wantErrorCount)
+			}
+			wantStatuses := make([]*InvocationCommentStatusRecord, 0)
+			if tc.wantErrorCount == 0 {
+				wantStatuses = tc.statuses
+			}
+			if diff := cmp.Diff(wantStatuses, tc.bqClient.processedStatuses); diff != "" {
+				t.Errorf("unexpected mismatch (-want +got):\n%s", diff)
+			}
```

Comment:
I think `cmp.Diff` should be able to handle comparing two empty slice so you don't need lines 300-303

Reply:
tc.statuses is not "gotStatuses". The insert behavior of the fake client is to store the input statuses in memory. But if the fake client is stubbed with an insertion error, it won't store the statuses and will return the stubbed error. I use `tc.wantErrorCount` to check for that since it's a simple proxy for expecting the error case.